email_reply_parser/__init__.py

```diff
@@ -94,18 +94,20 @@ def _scan_line(self, line):
 
             line - a row of text from an email message
         """
-        is_quote_header = self.QUOTE_HDR_REGEX.match(line) is not None
-        is_quoted = self.QUOTED_REGEX.match(line) is not None
-        is_header = is_quote_header or self.HEADER_REGEX.match(line) is not None
+        stripped_line = line.strip()
```

Comment:
What do you think about changing `HEADER_REGEX` from `r'^\*?(From|Sent|To|Subject):\*? .+'` to `r'^\s*\*?(From|Sent|To|Subject):\*? .+'` instead?
Other expressions seem not to care about leading whitespaces

Reply:
I saw that Yahoo! started behaving weirdly - they add extra spaces everywhere, then they add what appear to be random `|`s where the original message should be (are they trying to recreate the table cells using `|`s?). See for example:
```
 |

|
|
| New Message from Alexandru on Sailo |

 |

 |

|
|
| Ahoy Alexandru, |

 |

 |

|
|
|  Alexandru has sent you a message regarding a trip aboard the X-Yachts Xp 38. |

 |

 |
```

I figured it was safer, for parsing purposes, if we ignored trailing spaces on every line.

Reply:
It may result in working incorrectly when someone adds trailing leeading spaces intentionally.
For example, list items or code with indents.

I agree that there is a little chance that the spaces will be important in our use case, but for a general-purpose library it's not good

Reply:
I agree an email may contain whitespace that's no accident. However, you're gonna have a hard time distinguishing between user intent and email servers' vagaries.

I updated the code to reflect the change that's been observed in Yahoo! behavior. Indeed, it's fair that I don't make assumptions about the other lines.